subprojects/execution/src/main/java/org/gradle/internal/execution/impl/steps/CacheStep.java

```diff
@@ -56,6 +56,9 @@ public CacheStep(
 
     @Override
     public CurrentSnapshotResult execute(C context) {
+        if (!buildCache.isEnabled()) {
+            return executeWithoutCache(context);
+        }
```

Comment:
I omitted this intentionally, and I don't think we should add it now.

Reply:
What is the reason you omitted the check for caching enabled in the `CacheStep`? I don't see a reason to try to cache things if there is no build cache to use. What is the advantage that all the units of work need to implement this logic as well?

Reply:
The logic of ignoring the cache if it's not enabled should be built into `context.getCacheHandler()`.

We could introduce different kinds of `UnitOfWork` (like `CacheableUnitOfWork`) if we wanted to have work items that are not cacheable (and won't be in the foreseeable future), but that didn't seem like a necessity at this point.